src/master-playlist-controller.js

```diff
@@ -1286,34 +1302,36 @@ export class MasterPlaylistController extends videojs.EventTarget {
    */
   excludeIncompatibleVariants_(media) {
     let codecCount = 2;
-    let videoCodec = null;
-    let codecs;
+    let codecs = {video: {}};
 
     if (media.attributes.CODECS) {
       codecs = parseCodecs(media.attributes.CODECS);
-      videoCodec = codecs.videoCodec;
-      codecCount = codecs.codecCount;
+      codecCount = Object.keys(codecs).length;
     }
 
     this.master().playlists.forEach(function(variant) {
-      let variantCodecs = {
-        codecCount: 2,
-        videoCodec: null
-      };
+      let variantCodecCount = 2;
+      let variantCodecs = {video: {}};
 
       if (variant.attributes.CODECS) {
         variantCodecs = parseCodecs(variant.attributes.CODECS);
+        variantCodecCount = Object.keys(variantCodecs).length;
+      }
+
+      // The number of streams cannot change
+      if (variantCodecCount !== codecCount) {
+        variant.excludeUntil = Infinity;
       }
 
-      // if the streams differ in the presence or absence of audio or
-      // video, they are incompatible
-      if (variantCodecs.codecCount !== codecCount) {
+      // the video codec cannot change
+      if (variantCodecs.video && codecs.video &&
+        variantCodecs.video.type !== codecs.video.type) {
         variant.excludeUntil = Infinity;
       }
 
-      // if h.264 is specified on the current playlist, some flavor of
-      // it must be specified on all compatible variants
-      if (variantCodecs.videoCodec !== videoCodec) {
+      // the audio codec cannot change
```

Comment:
previously we ignored the audio codec all together, seemed wrong to me.

Reply:
I think you're right, but it may be safer to isolate that change to a separate PR. Unless we know that browsers can't switch audio codecs.

Reply:
This was one of the failures I encountered on edge. We setup source buffers with `avc` video  and `aac`. Then we try to use a playlist that has `avc` video and `ac-3` audio because we didn't blacklist as we were not changing types.

Reply:
👍 all good then